tokio-trace/src/dispatcher.rs

```diff
@@ -0,0 +1,37 @@
+pub use tokio_trace_core::dispatcher::*;
+
+use std::cell::RefCell;
+
+thread_local! {
+    static CURRENT_DISPATCH: RefCell<Dispatch> = RefCell::new(Dispatch::none());
+}
+
+/// Sets this dispatch as the default for the duration of a closure.
+///
+/// The default dispatcher is used when creating a new [`Span`] or
+/// [`Event`], _if no span is currently executing_. If a span is currently
+/// executing, new spans or events are dispatched to the subscriber that
+/// tagged that span, instead.
+///
+/// [`Span`]: ::span::Span
+/// [`Subscriber`]: ::Subscriber
+/// [`Event`]: ::Event
+pub fn with_default<T>(dispatcher: Dispatch, f: impl FnOnce() -> T) -> T {
+    let prior = CURRENT_DISPATCH.try_with(|current| current.replace(dispatcher));
+    let result = f();
```

Comment:
Should this use some drop guard so that the prior dispatcher is set back even if the closure panics?

Reply:
Switched to use a drop guard in 9c38d67074d2d24e928c4fc52ddae8186904edf9. Good call!